explainability-service/src/test/java/org/kie/trustyai/service/scenarios/batching/TestBatching.java

```diff
@@ -0,0 +1,153 @@
+import io.quarkus.test.junit.TestProfile;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+@QuarkusTest
+@TestProfile(BatchingTestProfile.class)
+class TestBatching {
+
+    private static final String makeHeader(List<String> names) {
+        return String.join(",", names.stream().map(name -> "\"" + name + "\"").collect(Collectors.toList()));
+    }
+
+    @Test
+    void testFewerThanBatch() throws IOException {
+        final int batchSize = 20;
+        final int N = 10;
+        final List<String> names = List.of("xa-1", "xa-2", "xa-3");
+        final BatchingStorage batchStorage = new BatchingStorage();
+        batchStorage.setNames(names);
+        batchStorage.setObservations(N);
+        final InputStream stream = batchStorage.getDataInputStream("inputs.csv");
+        final List<String> lines = BatchReader.readEntries(stream, batchSize);
+        System.out.println(lines);
+        assertEquals(N + 1, lines.size());
+        assertEquals(makeHeader(names), lines.get(0));
+    }
+
+    @Test
+    void testBiggerThanBatch() throws IOException {
+        final int batchSize = 20;
+        final int N = 1000;
+        final List<String> names = List.of("xa-1");
+        final BatchingStorage batchStorage = new BatchingStorage();
+        batchStorage.setNames(names);
+        batchStorage.setObservations(N);
+        final InputStream stream = batchStorage.getDataInputStream("inputs.csv");
+        final List<String> lines = BatchReader.readEntries(stream, batchSize);
+        System.out.println(lines);
+        assertEquals(batchSize + 1, lines.size());
+        assertEquals(makeHeader(names), lines.get(0));
+    }
+
+    @Test
+    void testConvertToDataframe() throws IOException {
+        final int batchSize = 20;
+        final int N = 1000;
+        final List<String> names = List.of("xa-1");
+        final BatchingStorage batchStorage = new BatchingStorage();
+        batchStorage.setNames(names);
+        batchStorage.setObservations(N);
+        final InputStream inStream = batchStorage.getDataInputStream("inputs.csv");
+        final List<String> inLines = BatchReader.readEntries(inStream, batchSize);
+        final ByteBuffer inputBuffer = ByteBuffer.wrap(BatchReader.linesToBytes(inLines));
+
+        final InputStream outStream = batchStorage.getDataInputStream("outputs.csv");
+        final List<String> outLines = BatchReader.readEntries(outStream, batchSize);
+        final ByteBuffer outputBuffer = ByteBuffer.wrap(BatchReader.linesToBytes(outLines));
+
+        final CSVParser parser = new CSVParser();
+        final Dataframe dataframe = parser.toDataframe(inputBuffer, outputBuffer);
+
+        assertEquals(batchSize, dataframe.getRowDimension());
+        assertEquals(names, dataframe.getInputDataframe().getColumnNames());
+        assertEquals(names, dataframe.getOutputDataframe().getColumnNames());
+        assertEquals(2, dataframe.getColumnDimension());
+    }
```

Comment:
Can we add an assertion here that the recovered batch is indeed returning the first 20 lines of the csv/make a more robust test that ensures each batch is indeed grabbing the data we think it is?

Reply:
@RobGeada Since the values are increasing, I've added a check that they are the expected ones.